packages/headless/src/modules/brands/repositories/index.ts

```diff
@@ -0,0 +1,35 @@
+import { Product } from '../../../entity/product';
+import { IBrandDatabase } from './brand.database.interface';
+import { Injectable } from '@nestjs/common';
+import { Brand } from 'src/entity/brand';
+
+@Injectable()
+export class BrandRepository {
+    constructor(private readonly db: IBrandDatabase) { }
+
+    async findBrandById(brandId: string): Promise<Brand | null> {
+        const brand = await this.db.getBrandById(brandId);
+        return brand;
+    }
+
+    async findAllBrands(skip?: number, limit?: number): Promise<Brand[]> {
+        const brands = await this.db.getAllBrands(skip, limit);
+        return brands;
+    }
+
+    async createBrand(brand: Brand): Promise<Brand | null> {
+        const savedBrand = await this.db.addNewBrand(brand);
+        return savedBrand;
+    }
+
+    async updateBrandById(brandId: string, brandUpdates: Brand): Promise<Brand | null>{
```

Comment:
Do the same thing as suggested before (naming convention)

Reply:
if we need to search anything based on other criterias, then would have to rename the query paramter as well, but for now the api is to find by ID, that is why it is set like that

Reply:
Do the same thing as suggested before